### ISSUE TITLE
NoMethodError - undefined method `to_sym' for nil:NilClass

### ISSUE THREAD

**Issue by araslanov-e** [rubysherpas/paranoia#169]
bundle `gem "paranoia", github: "radar/paranoia", branch: "rails4"`

```
User.all

NoMethodError - undefined method `to_sym' for nil:NilClass:
  activerecord (4.1.5) lib/active_record/sanitization.rb:59:in `block in expand_hash_conditions_for_aggregates'
  activerecord (4.1.5) lib/active_record/sanitization.rb:58:in `expand_hash_conditions_for_aggregates'
  activerecord (4.1.5) lib/active_record/relation/query_methods.rb:944:in `build_where'
  activerecord (4.1.5) lib/active_record/relation/query_methods.rb:572:in `where!'
  activerecord (4.1.5) lib/active_record/relation/query_methods.rb:559:in `where'
  activerecord (4.1.5) lib/active_record/querying.rb:10:in `where'
   () .../bundler/gems/paranoia-ba0e7b920239/lib/paranoia.rb:185:in `block in acts_as_paranoid'
  activerecord (4.1.5) lib/active_record/scoping/default.rb:103:in `block (3 levels) in build_default_scope'
  activerecord (4.1.5) lib/active_record/relation.rb:285:in `scoping'
  activerecord (4.1.5) lib/active_record/scoping/default.rb:103:in `block (2 levels) in build_default_scope'
  activerecord (4.1.5) lib/active_record/scoping/default.rb:102:in `block in build_default_scope'
  activerecord (4.1.5) lib/active_record/scoping/default.rb:125:in `evaluate_default_scope'
  activerecord (4.1.5) lib/active_record/scoping/default.rb:101:in `build_default_scope'
  activerecord (4.1.5) lib/active_record/scoping/named.rb:33:in `default_scoped'
  activerecord (4.1.5) lib/active_record/scoping/named.rb:28:in `all'
  activerecord (4.1.5) lib/active_record/querying.rb:10:in `where'
```

**Comment by radar** [comment]
Please provide exact steps to reproduce this issue.

**Comment by araslanov-e** [comment]
Set gem to Gemfile

```
$ bundle
$ bin/rails c
$ => User.all
$ => NoMethodError - undefined method `to_sym' for nil:NilClass
```

all

**Comment by radar** [comment]
I cannot reproduce this problem in a new app. Please upload an app to GitHub which reproduces this problem.

**Comment by deevis** [comment]
Hey guys, I'm getting this same error.  It is coming from 

```
activerecord-4.1.6/lib/active_record/sanitization.rb:59:in `block in expand_hash_conditions_for_aggregates
```

Does that shine any light on things?

Also - I'm having the problem on a many-to-many class that DOESN'T have a primary key.  This seems to be the root of the problem.  

paranoia#touch_paranoia_column  ends up calling active_record/persistence#touch which then needs a primary_key and, well - we don't have a primary key.

So - do I need to have a primary key?  DId I forget to RTFM?  I will now go RTFM...

**Comment by bluej100** [comment]
Reproduction based on help from @deevis :

https://github.com/bluej100/paranoia/commit/2428210ec3eec282ecb74cc47652a88976b70c55

I don't think it's worth supporting tables without primary keys, since ActiveRecord doesn't really support them (without https://github.com/composite-primary-keys/composite_primary_keys ) and they're easy to add.

**Comment by deevis** [comment]
So, yeah, I had previously added the Primary Keys and can confirm the problem definitely goes away.

Adding the Primay Key column was trivial - but deciding I needed to do so was, well, this entire process...

So, just suggesting this Primary Key dependency make it into the general info about how Paranoia can (and can't) be used.

And thanks for an awesome project!

**Comment by bluej100** [comment]
Submitted a pull request for an error when trying to use acts_as_paranoid on models without primary keys.

**Comment by radar** [comment]
Fixed by #178.

**Comment by jhawthorn** [comment]
Sorry, I've had to revert #178, because of causing issues before a database connection is available (see #186). We 

I don't believe a warning belongs in this gem. The error occurs within ActiveRecord's touch, and that's where the warning should be.
